Initial contrib/packaging.

nsextonTT opened the pull request:
### Ticket
Closes https://github.com/tenstorrent/tt-metal/issues/27557

### Problem description
We need a place to put the distro enablement bounties.

### What's changed
* Added a contrib/packaging folder to organize community-level supported distros.
* DevRel will be taking point on the effort.
* Added a README to outline what this new folder is and isn't (please review). 
* Pulled in debian 12 bounty so we can finalize the bounty. Thank you [tiamoesg](https://github.com/tiamoesg)! https://github.com/tenstorrent/tt-metal/issues/18297  https://github.com/tenstorrent/tt-metal/pull/25922
* Pulled in QubesOS 4 bounty so we can finalize the bounty. Thank you [silvanshade](https://github.com/silvanshade)! https://github.com/tenstorrent/tt-metal/issues/18293 https://github.com/tenstorrent/tt-metal/pull/27872

### Checklist
N/A

tiamoesg committed:
parent 792094e77f943668266790936b3cd335863b80ab
author Tiamoesg <tiamoesg@gmail.com> 1753741374 -0500
committer nsextonTT <216619413+nsextonTT@users.noreply.github.com> 1756192370 -0400

Create install_tt-debian.sh

Automated installer for TT on Debian 12

Update build_metal.sh

Added Debian 12 compatibility

Update create_venv.sh

Added formatting and exports for environment variables. Including device selection

Update create_stubs.sh

Added formatting to match install scripts

Update install_tt-debian.sh

replaced curl commands that went to my custom repo scripts. -- now that the build_metal.sh and install_dependencies.sh will be proper; they aren't needed.

Update create_venv.sh

deleted duplicate line and declared red

Update create_venv.sh

Color definitions updated

Update create_venv.sh

Added logging functions to avoid in-band colorization.

Fixed "pre-commit install" to use python_venv, to not require"source"ing the script.

Update install_tt-debian.sh

Added logging functions to avoid in-band colorization

Update create_stubs.sh

Added logging functions to avoid in-band colorization

nsextonTT committed:
Initial skeleton for contrib/packaging.

marty1885 reviewed (approved):
@nsextonTT Could/Should I also be on the reviewers list? I am currently the maintainer of our (unofficial) AUR packages. Maybe I can provide input for related distros if they show up?

nsextonTT committed:
Adding JHawley to packaging codeowners

tsingletaryTT commented on contrib/packaging/README.md:
```diff
@@ -0,0 +1,28 @@
+# Community-Driven Linux Distribution Support
+
+Welcome to the community-driven Linux distribution support folder for Tenstorrent's tt-metal repository! This folder is dedicated to providing scripts, installers, packaging, and guides for getting Tenstorrent software and hardware working on various Linux distributions beyond what Tenstorrent officially supports. While Tenstorrent does not officially support these distributions, we believe in fo ...
+
```
Can we add a bullet list or a table of the currently community-supported distros? We can wait to do it until later iteratively too, but it'd be great to have a bird's eye view on the distros, version, gotchas, etc and save people a click-through to each next README.

tsingletaryTT reviewed (approved):
This is good as-is! We can iterate and make it better and a more unique showcase for community distro support.

nsextonTT committed:
Refactor dirs, update README, mv debian installer.

silvanshade committed:
Add Qubes OS install guide (#27872)

This PR adds a guide for installing the stack on Qubes OS.

This is a modified version of the original guide I posted in the bounty
thread with some editing for clarity and simplifications enabled through
patches I upstreamed to
[tenstorrent/tt-installer](https://github.com/tenstorrent/tt-installer).

Closes tenstorrent/tt-metal#18293

CC @nsextonTT

marty1885 committed:
[skip ci] Add AUR and arch enablment (#28909)

Adds Arch Linux as community supported distro

Currently I maintain most Tenstorrent related packages on AUR.
https://aur.archlinux.org/packages?K=tenstorrent . Including TTNN as an
AUR packages (takes a while to build.. but it works).

All versioned packages, besides TTNN, are automatically checked against
latest release. Updates and checked on my home machine daily.

<img width="3680" height="800" alt="image"
src="https://github.com/user-attachments/assets/b961d062-1bd3-428e-bb3b-4a3e4d2c793b"
/>

rekudyu committed:
Added alpine readme (#29494)

### Ticket
https://github.com/tenstorrent/tt-metal/issues/18290

### Problem description
Added Alpine installation documentation.

### What's changed
Adds alpine linux installation instructions.

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI with demo tests passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml)
CI passes (if applicable) See [recommended dev
flow](https://github.com/tenstorrent/tt-metal/blob/main/models/docs/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy
quick](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml)
CI passes (if applicable)
- [ ] [Galaxy demo tests, for
Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml)
CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml)
CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml)
CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

jbakerTT committed:
Improve packaging README legal language and fix CODEOWNERS errors

- Enhanced disclaimer with explicit warranty and liability language
- Added contributor certification requirements
- Added indemnification clause
- Clarified patent grant provisions in licensing section
- Removed redundant 'use at your own risk' statement
- Fixed CODEOWNERS syntax errors:
  - Added missing leading / to METALIUM_GUIDE.md
  - Added owners to tests/sweep_framework/sweeps/
  - Removed invalid glob pattern /models/*/**
  - Consolidated functional_*/ pattern
